### PR TITLE
v0.12.0a3

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
This includes backports of everything relevant from main.

### CODE REVIEW

**Review by marc-hb — APPROVED** [review]
I'm assuming all the cherry-picks are clean and rubber-stamping this.

**Comment by mbolivar-nordic** [comment]
> I'm assuming all the cherry-picks are clean and rubber-stamping this.

Yep, no conflicts.

File: src/west/configuration.py [hunk 1]
```diff
@@ -233,7 +233,7 @@ def _location(cfg: ConfigFile, topdir: Optional[PathType] = None) -> str:
         if 'BSD' in plat:
             return '/etc/westconfig'
 
-        if 'CYGWIN' in plat:
+        if 'CYGWIN' in plat or 'MSYS_NT' in plat:
```
Comment (tejlmand) [inline src/west/configuration.py:236]: nothing wrong on this line, but why is there no `signed-off-by` entry in the commit message ?

**Review by tejlmand — COMMENTED** [review]
Not a pure rubber stamping, I actually did a quick look through :smile:

**Pull request merged**